api/migrations/20200709183253-loadMemDocs.js

```diff
@@ -213,22 +216,36 @@ exports.up = async function (db) {
             } else {
               // check doc has valid collection
               if (!existingDoc.collectionId.equals(bcmiCollection._id)) {
-                console.log(`Found document ${existingDoc._id} with a bad collection id ${existingDoc.collectionId}, adding to proper collection: ${bcmiCollection._id}`)
+                console.log(`Found record ${existingDoc._id} with a bad collection id ${existingDoc.collectionId}, adding to proper collection: ${bcmiCollection._id}`)
+
                 existingDoc.collectionId = bcmiCollection._id;
                 await nrpti.findOneAndUpdate({ _id: existingDoc._id }, existingDoc);
-                // duplicate prevention
+                // duplicate record prevention
                 const arrayIncludes = bcmiCollection.records.some(item => item.equals(existingDoc._id));
                 if (!arrayIncludes) {
-                  console.log(`Adding doc to docsArray ${existingDoc._id}`);
+                  console.log(`Adding doc ${existingDoc._id} to docsArray of ${bcmiCollection._id}`);
                   allNewDocs.push(existingDoc._id);
                 }
               }
               docsExisting += 1;
+              // Ensure that records only exist in the proper collection for their mine
+              const collectionsToFix = await nrpti.find({records: { $in: [ existingDoc._id ]}});
+              collectionsToFix.forEach( async (coll) => {
+                if (!coll.project.equals(nrptiMine._id)) {
```

Comment:
Does this work. `.equals` is Java if I remember correctly. 
`(coll.project !== nrptiMine._id)`

Reply:
It should be an mongo objectid method, but I don't explicitly cast to obj id. Might just be strings.. It's also used for the existing collection id comparison on ln 218. I'll double check this

Reply:
This should be correct the, the strict comparison operator doesn't seem to work on ObjectIds. Despite being console logged as a string I think they are using the mongo object id in the script.